usr/lib/common/template.c

```diff
@@ -638,25 +636,25 @@ CK_RV template_unflatten_withSize(TEMPLATE **new_tmpl, CK_BYTE *buf,
                 TRACE_ERROR("%s\n", ock_err(ERR_HOST_MEMORY));
                 return CKR_HOST_MEMORY;
             }
-            a2->type = a1->type;
+            a2->type = a1.type;
 
-            if ((a1->type == CKA_CLASS || a1->type == CKA_KEY_TYPE
-                 || a1->type == CKA_MODULUS_BITS
-                 || a1->type == CKA_VALUE_BITS
-                 || a1->type == CKA_CERTIFICATE_TYPE
-                 || a1->type == CKA_VALUE_LEN)
-                && a1->ulValueLen != 0) {
+            if ((a1.type == CKA_CLASS || a1.type == CKA_KEY_TYPE
+                 || a1.type == CKA_MODULUS_BITS
+                 || a1.type == CKA_VALUE_BITS
+                 || a1.type == CKA_CERTIFICATE_TYPE
+                 || a1.type == CKA_VALUE_LEN)
+                && a1.ulValueLen != 0) {
 
                 a2->ulValueLen = sizeof(CK_ULONG);
 
                 {
-                    CK_ULONG_32 *p32;
+                    CK_ULONG_32 p32;
```

Comment:
another variable name like u32 would be better, the p is probably a leftover from pointer

Reply:
done